python/ray/data/impl/arrow_block.py

```diff
@@ -87,34 +98,42 @@ def slice(self, start: int, end: int, copy: bool) -> "pyarrow.Table":
 
     def random_shuffle(self, random_seed: Optional[int]) -> "pyarrow.Table":
         random = np.random.RandomState(random_seed)
-        return self._table.take(random.permutation(self.num_rows()))
+        return self.take(random.permutation(self.num_rows()))
 
     def schema(self) -> "pyarrow.lib.Schema":
         return self._table.schema
 
     def to_pandas(self) -> "pandas.DataFrame":
         return self._table.to_pandas()
 
-    def to_numpy(self, column: str = None) -> np.ndarray:
-        if column is None:
-            raise ValueError(
-                "`column` must be specified when calling .to_numpy() "
-                "on Arrow blocks."
-            )
-        if column not in self._table.column_names:
-            raise ValueError(
-                f"Cannot find column {column}, available columns: "
-                f"{self._table.column_names}"
-            )
-        array = self._table[column]
-        if array.num_chunks > 1:
-            # TODO(ekl) combine fails since we can't concat
-            # ArrowTensorType?
-            array = array.combine_chunks()
+    def to_numpy(
+        self, columns: Optional[Union[str, List[str]]] = None
+    ) -> Union[np.ndarray, Dict[str, np.ndarray]]:
+        if columns is None:
+            columns = self._table.column_names
```

Comment:
This change in semantics is a small leak from the larger PR, but shouldn't be observable in the Datasets API; callers of `.to_numpy()` should always pass a single column name in this branch. So I think that this being included is fine.